src/qr.rs

```diff
@@ -684,6 +684,11 @@ pub async fn set_config_from_qr(context: &Context, qr: &str) -> Result<()> {
                 .await?;
         }
         Qr::Socks5Proxy { host, port } => {
+            // disable proxy before changing settings to not use a combination of old and new
+            context
```

Comment:
Finally i think it's better to set `Socks5Host` to None here and to `Some(&host)` in the end, this way settings are invalid until configured and there's no chance to just drop the old settings

Reply:
hm, requiring some implicit ordering seems more error prone than surrounding by an disable/enable brace.

Reply:
The problem is that after a crash the socks5 settings might be just disabled, it's not what is expected, either old or new settings should be used. But as we don't have a way to change the settings atomically, i suggest to "invalidate" them. It's error-prone in the sense that Delta Chat might not work at all after restart, but it seems better than working w/o proxy at all. I think if a comment is added, it should be reliable enough

Reply:
Actually, there _is_ a way to change the settings atomically using a db transaction, but probably it will be a little more code

Reply:
chnaging several settings at atomically is also not done elsewhere and this discussion seems out of scope of this pr. and also a bit theoretical: the qr code was just scanned, the app is far away from being terminated by os (which would be the most common case for incomplete settings)

i think, the current state is as good as other parts, not sure if spending more energy here currently is worth while there are bigger fishes to get :)